s3/replication/build-runalltest.sh

```diff
@@ -0,0 +1,192 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+#
+# build-runalltest.sh allows to build all multisite moudles,
+# create virtual environment, populating service files with
+# correct paths, copying to default systemd location and
+# starting daemon processes and running load transfer tests for
+# s3replicator and s3replicationmanager.
+#
+# Usage : sh ./build-runalltest.sh
+# 
+
+
+# Project root directory
+MULTISITE_ROOT_DIR="$(cd ../../; pwd)"
+echo "Project directory : "$MULTISITE_ROOT_DIR
+
+# Replication directory
+REPLICATION_DIR="$(pwd)"
+echo "Replication directory :" $REPLICATION_DIR
+
+# Module directory
+S3_REPLICATION_COMMON_SRC_DIR=$REPLICATION_DIR/common/src
+S3_REPLICATION_MANAGER_SRC_DIR=$REPLICATION_DIR/manager/src
+S3_REPLICATOR_SRC_DIR=$REPLICATION_DIR/replicator/src
+
+# Test directory
+S3_REPLICATION_COMMON_TEST_DIR=$REPLICATION_DIR/common/tests/system
+S3_REPLICATION_MANAGER_TEST_DIR=$REPLICATION_DIR/manager/tests/system
+S3_REPLICATOR_TEST_DIR=$REPLICATION_DIR/replicator/tests/system
+
+echo "Module directories -"
+echo "Common module directory :" $S3_REPLICATION_COMMON_SRC_DIR
+echo "Manager module directory :" $S3_REPLICATION_MANAGER_SRC_DIR
+echo "Replicator module directory :" $S3_REPLICATOR_SRC_DIR
+
+# Active an environment
+python3 -m venv $REPLICATION_DIR/s3env
+source $REPLICATION_DIR/s3env/bin/activate
+
+# Check and download dependencies
+pip3 install --upgrade setuptools
+
+pip3 install -r $REPLICATION_DIR/common/requirements.txt
+pip3 install -r $REPLICATION_DIR/manager/requirements.txt
+pip3 install -r $REPLICATION_DIR/replicator/requirements.txt
+
+cd $REPLICATION_DIR
+make clean install
+
+# Create and copy files
+S3_IN_MEMRORY_SVC_FILE=$S3_REPLICATION_COMMON_SRC_DIR/systemd/s3_in_memory.service
+MANGER_SVC_FILE=$S3_REPLICATION_MANAGER_SRC_DIR/systemd/s3replicationmanager.service
+REPLICATOR_SVC_FILE=$S3_REPLICATOR_SRC_DIR/systemd/s3replicator.service
+
+# Systemd service files
+SYSD_S3_IN_MEMORY_SVC_FILE=/etc/systemd/system/s3_in_memory.service
+SYSD_REPLICATION_MANAGER_SVC_FILE=/etc/systemd/system/s3replicationmanager.service 
+SYSD_REPLICATOR_SVC_FILE=/etc/systemd/system/s3replicator.service
+
+# Create creadential file in home directory with below path.
+# ~/.cortxs3/credentials.yaml
+CORTX_S3_CRED_FILE=~/.cortxs3/credentials.yaml
+CORTX_S3_DIR="$(dirname "$CORTX_S3_CRED_FILE")"
+if [ -f "$CORTX_S3_CRED_FILE" ]; then
+    echo "$CORTX_S3_CRED_FILE exist!"
+else
+    if [ -d $CORTX_S3_DIR]; then
+        echo "$CORTX_S3_DIR exist!"
+    else
+        mkdir $CORTX_S3_DIR
+    fi
+    touch $CORTX_S3_CRED_FILE
+    echo "access_key: '<your access_key>'" > $CORTX_S3_CRED_FILE
```

Comment:
Add comment.
Placeholder for keys to run against in-memory s3 server.

Reply:
done